src/baldr/graphreader.cc

```diff
@@ -630,5 +645,43 @@ std::unordered_set<GraphId> GraphReader::GetTileSet(const uint8_t level) const {
   return tiles;
 }
 
+AABB2<PointLL> GraphReader::GetMinimumBoundingBox(const AABB2<PointLL>& bb) {
```

Comment:
this is one of the main areas of new functionality. will give you a right sized bounding box that will look at the input box, find all the nodes in it and expand/contract the bounding box to fit the edges leaving those nodes and their shape